presto-main/src/main/java/io/prestosql/Session.java

```diff
@@ -331,7 +356,7 @@ public Session beginTransactionId(TransactionId transactionId, TransactionManage
                     .orElseThrow(() -> new PrestoException(NOT_FOUND, "Catalog does not exist: " + catalogName))
                     .getConnectorId();
             if (role.getType() == SelectedRole.Type.ROLE) {
-                accessControl.checkCanSetRole(transactionId, identity, role.getRole().get(), catalogName);
+                accessControl.checkCanSetRole(sessionWithTransaction, role.getRole().get(), catalogName);
```

Comment:
revert. instead pass `identity` to `checkCanSetRole `

Reply:
I agree, constructing the session twice seems like a hack.